src/main_test.go

```diff
@@ -15,9 +15,44 @@
 package main
 
 import (
+	"reflect"
 	"testing"
+
+	"github.com/coreos/fuze/third_party/github.com/coreos/ignition/config"
+	"github.com/coreos/fuze/third_party/github.com/go-yaml/yaml"
 )
 
-func Test(t *testing.T) {
-	// TODO
+func TestHasUnrecognizedKeys(t *testing.T) {
+	tests := []struct {
+		in    string
+		unrec bool
+	}{
+		{
+			in:    "ignition_version: 1",
+			unrec: false,
+		},
+		{
+			in:    "ignition_version: 1\npasswd:\n users:\n  - name: foobar\n",
+			unrec: false,
+		},
+		{
+			in:    "foo: bar",
```

Comment:
IMO might be nice to return a more specific error if `ignition_version` is missing since in principle that field is what defines it as an ignition configuration and not just some random json. Just a thought.